### ISSUE TITLE
Fix `selector-max-type` false positives with nested selectors

### ISSUE THREAD

**Issue by sandersky** [stylelint/stylelint#9318]
### What minimal example or steps are needed to reproduce the bug?

```css
.foo {
  table {
    color: red;
  }
}
```

### What minimal configuration is needed to reproduce the bug?

```json
{
  "rules": {
    "selector-max-type": [0, { "ignore": ["child", "compounded", "descendant", "next-sibling"] }]
  }
}
```

### How did you run Stylelint?

https://stylelint.io/demo/#N4Igxg9gJgpiBcIB0AzCEAEwA6A7DGALgIYBGANjFngQZORAE7waMxQDcNGAvnjyAA0IFAEtKAOWIBbOIhgAPGQAdKSMAGcNQ8BFxiA5ghCLlTQhlgpiAV3IWc+DIsIxcUDSwDaAcg2EAT0pyUVxCAFpIfVEDcP9id2JGKB8AXUFuRjsYT2onbBANGEowQiZw6WIFcMDlGALvAAZBLAwY3CYYbwKwAAtxKAKWnohpMxt3diGMAtgNMDcoBMJpgtwXONEKUIMC1N5U7j5cHi5cHSjDADEmSsJjfyDi0MJ1PUMkaQArbWFYZW0iEcBAKj2CLwaMxA5GIrn8Q24oMC4LCkXeMTiJESyUhBRhcJWIH4eBAPCAA

### Which Stylelint-related dependencies are you using?

```json
{
  "stylelint": "17.12.0",
  "stylelint-config-standard": "40.0.0"
}
```

### What did you expect to happen?

I'd expect the rule to not complain about the nested `table` because it is effectively a descendent and I explicitly told the rule to ignore those.

### What actually happened?

```
2:3-8 error Too many type selectors in "table", maximum 0 [(selector-max-type)](https://stylelint.io/user-guide/rules/selector-max-type)
```

### Do you have a proposal to fix the bug?

The rule should ignored nested selectors when the parent selector involves something such as a classname or id.

**Comment by ybiquitous** [comment]
@sandersky Thanks for the report with the reproducible demo. I confirmed it's a false positive bug, as you think.

See also [another minimal demo](https://stylelint.io/demo/#N4Igxg9gJgpiBcIB0AzCEAEwA6A7DGALgIYBGANjFgL54EBkRZlNdGAfExVcLfho04lurXHzyp0XFrwlpMQ5jz4gANCBQBLSgDliAWziIYADwMAHSkjABnG2vARcWgOYIQp8xABOhDLBRiAFdyPxx+bxCYG3gsNgJsEBsYSjBCHwBafWITDMIAT3MYRNiAbQwABlUsDE0XXB8YMoxEsAALbShE6sTYGzAYXChiXEJEjABdDGpJtj5qAG48B0hnOoAxH2zCdxsCynJNUesnVyR9ACt7dVhze0RwhKT9lKOxhBaQAEYAdiQvgBMSAqiTwKmoQA):

```css
.foo {
  table {}      /* NG: Too many type selectors in "table", maximum 0 */
  & table {}    /* OK */
  > table {}    /* OK */
  & > table {}  /* OK */
}

.foo table {}   /* OK */
.foo > table {} /* OK */
```

---

I've labeled the issue as ready to implement. Please consider [contributing](https://stylelint.io/contributing) if you have time.

There are [steps on how to fix a bug in a rule](https://stylelint.io/contributor-guide/rules#fix-a-bug-in-a-rule) in the Contributor guide.
